bazel/src/main/java/com/google/api/codegen/bazel/ApisVisitor.java

```diff
@@ -61,6 +62,13 @@ interface FileWriter {
     this.srcDir = srcDir.normalize();
     this.destDir = destDir.normalize();
     this.overwrite = overwrite;
+    // Indicates a transport was supplied via command line and it should be respected.
+    // If false, any existing transport value pulled by buildozer will take precedent.
+    this.forceTransport = transport != null && !transport.isEmpty();
+    // Default to dual transport if command line did not indicate.
+    if (transport == null) {
+      transport = "grpc+rest";
```

Comment:
If I'm understanding correctly, this default is only useful if we are generating for the very first time, if we are regenerating and there is no transport passed in from command line, then this default will be overridden later by values from `buildozer`? If that's the case, do we have a flag indicating if we are generating for the first time or not? If we do, it would be better than the new `forceTransport` flag.
We could also override the transport in later steps only if both transport if null for both `buildozer` and command line.

Reply:
> We don't have a central place to define defaults like this, do we?

No, not really. As per the diff, it was previously set where the flags are parsed, which is a reasonable spot I guess, but I'd need to add a second variable there to track if the flag was set or not and pass that through to `ApiVisitor`. That would be fine too, but I think Blake's suggestion is a good one.

> do we have a flag indicating if we are generating for the first time or not

We don't have that today and I like the idea. I wonder how that would interact with automation/SAM, since I'm not sure it keeps track of if this is the first generation or not, so requiring that it provide `--first-generation` or something could be tricky. I will go find out and come back to this. I agree that would be a simpler approach.

Reply:
On the defaults: I do think it's worth keeping the defaults close together so whenever we need to change them we can easily find them.

I'm OK with the `--first-generation` idea conceptually. But if it makes things trickier to call build-gen, then I'd be inclined to make the external interface simple and contain the complexity within this tool.

Reply:
> I'm OK with the --first-generation idea conceptually. But if it makes things trickier to call build-gen, then I'd be inclined to make the external interface simple and contain the complexity within this tool.

I'm of two minds:
* trying to "guess" the use case the user is exercising (e.g. first gen, or regen with some changes) makes it harder for the tool to get things right for the user (writing this PR felt a _tad_ dirty)
* or we make it more explicit at the price of more "up front" cost to the user

I think at this point, since the process is already burdensome to users, I'd prefer to stick with the status quo, even if it makes this tool more complex in the short term. build_gen v2 should improve upon this experience (partly by being mostly automated!)

LMK if you feel strongly the other way in favor of a `--first-generation` flag.

Reply:
> I'd prefer to stick with the status quo

SGTM. It would be great if we can easily "guess" the use case based on the existing input, if not, creating a cumbersome/hack PR or add a new flag is not worth it for me.

Reply:
@vchudnov-g I moved the `transport` defaulting logic back to ArgsParser, and added a `forceTransport` boolean that it passes along to ApisVisitor via constructor. This is better. 

Thanks folks for the discussion.